crates/kubelet/src/kubelet.rs

```diff
@@ -162,6 +142,120 @@ impl<P> Clone for Kubelet<P> {
     }
 }
 
+/// Awaits SIGINT and sets graceful shutdown flag if detected.
+async fn start_signal_task(signal: Arc<AtomicBool>) -> anyhow::Result<()> {
+    ctrl_c().await?;
+    warn!("Caught keyboard interrupt.");
+    signal.store(true, Ordering::Relaxed);
```

Comment:
nit: it'd be good to have a comment here on ordering and why `Relaxed` is the right one.

Reply:
Well, my reasoning isn't very specific. We aren't trying to synchronize anything here. Once the flag is set, other tasks discover this with best effort and we don't really need any hard requirements on when this happens.

Reply:
I think this could also be implemented with a Mutex<bool> or something and ordering would be a non-issue.